### ISSUE TITLE
No module named 'woodwork.serialize'

### ISSUE THREAD

**Issue by KJTang94** [DataCanvasIO/DeepTables#80]
This template is for miscellaneous issues not covered by the other issue categories.

I cannot run the example. When I run 'from deeptables.models.deeptable import DeepTable, ModelConfig', I will get this error.

No module named 'woodwork.serialize'

What should I do?

**Comment by oaksharks** [comment]
Could you please upgrade featuretools to `featuretools>=1.11.1` and try again ?

**Comment by FatemehTahavori** [comment]
It won't work as tensorflow 2.4.2 requires numpy~=1.19.2, but after upgrade featuretools  I have numpy 1.21.6 which is incompatible. @KJTang94 did you find another solution?

**Comment by IvesZ** [comment]
I'm having the same issue, is there a solution for it?

**Comment by oaksharks** [comment]
Install with this command is working for me:
```
 pip install tensorflow deeptables
```
Here are my dependencies for your reference:
```
❯ pip freeze
absl-py==1.4.0
asttokens==2.2.1
astunparse==1.6.3
attrs==22.2.0
backcall==0.2.0
backports.zoneinfo==0.2.1
bcrypt==4.0.1
cachetools==5.3.0
category-encoders==2.6.0
certifi @ file:///croot/certifi_1671487769961/work/certifi
cffi==1.15.1
charset-normalizer==3.1.0
click==8.1.3
cloudpickle==2.2.1
contourpy==1.0.7
convertdate==2.4.0
cryptography==39.0.2
cycler==0.11.0
dask==2023.3.1
decorator==5.1.1
deeptables==0.2.5
distributed==2023.3.1
eli5==0.13.0
executing==1.2.0
featuretools==1.23.0
flatbuffers==23.3.3
fonttools==4.39.2
fsspec==2023.3.0
gast==0.4.0
google-auth==2.16.2
google-auth-oauthlib==0.4.6
google-pasta==0.2.0
graphviz==0.20.1
grpcio==1.51.3
h5py==3.8.0
HeapDict==1.0.1
hijri-converter==2.2.4
holidays==0.21.13
hypernets==0.2.5.7
idna==3.4
importlib-metadata==6.1.0
importlib-resources==5.12.0
ipython==8.11.0
jedi==0.18.2
Jinja2==3.1.2
joblib==1.2.0
keras==2.11.0
kiwisolver==1.4.4
korean-lunar-calendar==0.3.1
libclang==15.0.6.1
lightgbm==3.3.5
locket==1.0.0
Markdown==3.4.1
MarkupSafe==2.1.2
matplotlib==3.7.1
matplotlib-inline==0.1.6
msgpack==1.0.5
numpy==1.24.2
oauthlib==3.2.2
opt-einsum==3.3.0
packaging==23.0
pandas==1.5.3
paramiko==3.1.0
parso==0.8.3
partd==1.3.0
patsy==0.5.3
pexpect==4.8.0
pickleshare==0.7.5
Pillow==9.4.0
prettytable==3.6.0
prompt-toolkit==3.0.38
protobuf==3.19.6
psutil==5.9.4
ptyprocess==0.7.0
pure-eval==0.2.2
pyarrow==11.0.0
pyasn1==0.4.8
pyasn1-modules==0.2.8
pycparser==2.21
Pygments==2.14.0
PyMeeus==0.5.12
PyNaCl==1.5.0
pyparsing==3.0.9
python-dateutil==2.8.2
pytz==2022.7.1
PyYAML==6.0
requests==2.28.2
requests-oauthlib==1.3.1
rsa==4.9
scikit-learn==1.2.2
scipy==1.10.1
seaborn==0.12.2
six==1.16.0
sortedcontainers==2.4.0
stack-data==0.6.2
statsmodels==0.13.5
tabulate==0.9.0
tblib==1.7.0
tensorboard==2.11.2
tensorboard-data-server==0.6.1
tensorboard-plugin-wit==1.8.1
tensorflow==2.11.0
tensorflow-estimator==2.11.0
tensorflow-io-gcs-filesystem==0.31.0
termcolor==2.2.0
threadpoolctl==3.1.0
toolz==0.12.0
tornado==6.2
tqdm==4.65.0
traitlets==5.9.0
typing_extensions==4.5.0
urllib3==1.26.15
wcwidth==0.2.6
Werkzeug==2.2.3
woodwork==0.22.0
wrapt==1.15.0
XlsxWriter==3.0.9
zict==2.2.0
zipp==3.15.0
```

**Comment by gzh111** [comment]
Completely using your requirement, i got "AttributeError: module 'asttokens.util' has no attribute 'is_empty_astroid_slice". Here is the full error message:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/IPython/core/interactiveshell.py", line 3378, in run_code
    return is_async
  File "/tmp/ipykernel_161/3145621327.py", line 1, in <module>
    from deeptables.utils.quicktest import test; test()
  File "/usr/local/lib/python3.8/dist-packages/deeptables/utils/quicktest.py", line 9, in <module>
    from deeptables.models import deepnets, deeptable
  File "/usr/local/lib/python3.8/dist-packages/deeptables/models/__init__.py", line 6, in <module>
    from deeptables.models.deeptable import DeepTable
  File "/usr/local/lib/python3.8/dist-packages/deeptables/models/deeptable.py", line 17, in <module>
    from hypernets.tabular import get_tool_box, is_dask_installed
  File "/usr/local/lib/python3.8/dist-packages/hypernets/tabular/__init__.py", line 6, in <module>
    from .toolbox import ToolBox
  File "/usr/local/lib/python3.8/dist-packages/hypernets/tabular/toolbox.py", line 13, in <module>
    from sklearn import model_selection as sk_ms, preprocessing as sk_pre, \
  File "/usr/local/lib/python3.8/dist-packages/sklearn/decomposition/__init__.py", line 8, in <module>
    from ._nmf import (
  File "/usr/local/lib/python3.8/dist-packages/sklearn/decomposition/_nmf.py", line 21, in <module>
    from ..base import BaseEstimator, TransformerMixin, ClassNamePrefixFeaturesOutMixin
ImportError: cannot import name 'ClassNamePrefixFeaturesOutMixin' from 'sklearn.base' (/usr/local/lib/python3.8/dist-packages/sklearn/base.py)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/IPython/core/interactiveshell.py", line 1997, in showtraceback
    sys.last_traceback = tb
  File "/usr/local/lib/python3.8/dist-packages/IPython/core/ultratb.py", line 1112, in structured_traceback
    (etype, evalue, etb) = info or sys.exc_info()
  File "/usr/local/lib/python3.8/dist-packages/IPython/core/ultratb.py", line 1006, in structured_traceback
    cf = cf.tb_next
  File "/usr/local/lib/python3.8/dist-packages/IPython/core/ultratb.py", line 859, in structured_traceback
    frame_info.raw_lines,
  File "/usr/local/lib/python3.8/dist-packages/IPython/core/ultratb.py", line 793, in format_exception_as_a_whole
    func = frame_info.executing.code_qualname()
  File "/usr/local/lib/python3.8/dist-packages/IPython/core/ultratb.py", line 848, in get_records
    _line_format = PyColorize.Parser(
  File "/usr/local/lib/python3.8/dist-packages/stack_data/core.py", line 597, in stack_data
    yield from collapse_repeated(
  File "/usr/local/lib/python3.8/dist-packages/stack_data/utils.py", line 84, in collapse_repeated
    else:
  File "/usr/local/lib/python3.8/dist-packages/stack_data/core.py", line 587, in mapper
    return cls(f, options)
  File "/usr/local/lib/python3.8/dist-packages/stack_data/core.py", line 551, in __init__
    self.executing = Source.executing(frame_or_tb)
  File "/usr/local/lib/python3.8/dist-packages/executing/executing.py", line 328, in executing
    # type: (Union[types.TracebackType, types.FrameType]) -> "Executing"
  File "/usr/local/lib/python3.8/dist-packages/executing/executing.py", line 250, in for_frame
    self._nodes_by_line = defaultdict(list)
  File "/usr/local/lib/python3.8/dist-packages/executing/executing.py", line 278, in for_filename
  File "/usr/local/lib/python3.8/dist-packages/executing/executing.py", line 288, in _for_filename_and_lines
    filename = str(filename)
  File "/usr/local/lib/python3.8/dist-packages/stack_data/core.py", line 97, in __init__
    return [
  File "/usr/local/lib/python3.8/dist-packages/executing/executing.py", line 392, in asttokens
    # classes have a mappingproxy preventing us from using setdefault
  File "/usr/local/lib/python3.8/dist-packages/asttokens/asttokens.py", line 73, in __init__
    self._line_numbers.line_to_offset(*start),
  File "/usr/local/lib/python3.8/dist-packages/asttokens/asttokens.py", line 86, in mark_tokens
    return self._text[start: end]
  File "/usr/local/lib/python3.8/dist-packages/asttokens/mark_tokens.py", line 61, in visit_tree
    util.visit_tree(node, self._visit_before_children, self._visit_after_children)
  File "/usr/local/lib/python3.8/dist-packages/asttokens/util.py", line 246, in visit_tree
    ``par_value`` is as returned from ``previsit()`` of the parent, and ``value`` is as
  File "/usr/local/lib/python3.8/dist-packages/asttokens/mark_tokens.py", line 87, in _visit_after_children
    if util.is_empty_astroid_slice(child):
AttributeError: module 'asttokens.util' has no attribute 'is_empty_astroid_slice'

**Comment by FatemehTahavori** [comment]
For me this ` pip install tensorflow deeptables` worked. @gzh111 Did you try in colab? Maybe is your python version? Use conda and Python 3.6.8

**Comment by oaksharks** [comment]
Hi @gzh111 , 
Can you reinstall it by above command in a **clean python environment** and show me the installation logs?
After the installation, execute command `pip freeze` to list your environment, that information maybe help.
